### ISSUE TITLE
gocyclo min-complexity should be max-complexity

### ISSUE THREAD

**Issue by dnephin** [golangci/golangci-lint#368]
The value used to configure `gocyclo` is a maximum allowed complexity threshold. `min-complexity` is a bit confusing.

```
golangci-lint has version 1.12.5 built from 609de32 on 2018-12-23T09:42:24Z
```

**Comment by jirfag** [comment]
@mbana are you talking about `max-complexity`? it isn't implemented yet

**Comment by mbana** [comment]
@jirfag yes.
OK, thanks for confirming.

**Comment by jmatosp** [comment]
This is interesting what is the purpose of min-complexity? you want the code to have a minimal complexity?

Anyone using this?

**Comment by gurre** [comment]
This seems broken, when I'm using running `gocyclo -over 10 ./backend` I get several files with up to 15 in complexity. When I then run `golangci-lint run --config backend/.golangci.yml ./backend/...` with `enable-all: true` set in `.golangci.yml` it does not report any error at all.

**Comment by gurre** [comment]
I found the issue. `MinComplexity` is default 0 and not 10 as stated in the default config. `min-complexity` needs to be set in order for this linter to be enabled.

**Comment by gurre** [comment]
So it appears to be 30 that's default. It actually states 30 in a comment but then the value itself is different which I found confusion for an example configuration.

```
# minimal code complexity to report, 30 by default (but we recommend 10-20)
    min-complexity: 10
```

**Comment by mbana** [comment]
I am bit confused as well by this setting. Is it the minimal value at which the code is considered too complex or something else. If my understanding is correct, don't you think it should be called `max-complexity`?

A repository with demonstrating the effects of different values would be very useful at this point.

Thanks.

**Comment by jirfag** [comment]
@mbana  yes, it is the minimal value at which the code is considered too complex. It's the same as `--over` in `gocyclo`
